src/components/FindMovie/FindMovie.tsx

```diff
@@ -17,41 +68,81 @@ export const FindMovie: React.FC = () => {
               id="movie-title"
               placeholder="Enter a title to search"
               className="input is-dander"
+              value={query}
+              onChange={value => {
+                handleSearchChange(value);
+                setIsError(false);
+              }}
             />
           </div>
 
-          <p className="help is-danger" data-cy="errorMessage">
-            Can&apos;t find a movie with such a title
-          </p>
+          {isError && (
+            <p className="help is-danger" data-cy="errorMessage">
+              Can&apos;t find a movie with such a title
+            </p>
+          )}
         </div>
 
         <div className="field is-grouped">
           <div className="control">
             <button
               data-cy="searchButton"
               type="submit"
-              className="button is-light"
-            >
-              Find a movie
-            </button>
-          </div>
+              className={classNames(
+                'button',
+                'is-light',
+                { 'is-loading': !isLoaded },
+                { finally: isLoaded },
+              )}
+              disabled={queryLength === 0}
+              onClick={(event) => {
+                event.preventDefault();
 
-          <div className="control">
-            <button
-              data-cy="addButton"
-              type="button"
-              className="button is-primary"
+                setIsLoaded(false);
+
+                if (!query) {
+                  setIsLoaded(true);
+
+                  return;
+                }
+
+                handleClickFind();
+              }}
             >
-              Add to the list
+              {movie ? 'Search again' : 'Find a movie'}
             </button>
           </div>
+
+          {(movie && !isError) && (
+            <div className="control">
+              <button
+                data-cy="addButton"
+                type="button"
+                className="button is-primary"
+                onClick={() => {
+                  if (!isMovieDuplicate(movies, movie)) {
+                    addMovie(movie);
+                  }
+
+                  setMovie(null);
+                  setQuery('');
+                }}
+              >
+                Add to the list
+              </button>
+            </div>
+          )}
         </div>
       </form>
 
-      <div className="container" data-cy="previewContainer">
-        <h2 className="title">Preview</h2>
-        {/* <MovieCard movie={movie} /> */}
-      </div>
+      {movie && (
+        <div className="container" data-cy="previewContainer">
+          <h2 className="title">Preview</h2>
+          {movie && (
+            <MovieCard movie={movie} />
+          )}
+        </div>
+      )}
```

Comment:
```suggestion
      {movie && (
        <div className="container" data-cy="previewContainer">
          <h2 className="title">Preview</h2>
          {movie && (
            <MovieCard movie={movie} />
          )}
        </div>
      )}
```

why would you check for movie twice?